File: app/src/main/java/dev/ragnarok/fenrir/domain/IFeedbackInteractor.java
package dev.ragnarok.fenrir.domain;

import java.util.List;

import dev.ragnarok.fenrir.model.AnswerVKOfficialList;
import dev.ragnarok.fenrir.model.feedback.Feedback;
import dev.ragnarok.fenrir.util.Pair;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

public interface IFeedbackInteractor {
    Single<List<Feedback>> getCachedFeedbacks(int accountId);

    Single<Pair<List<Feedback>, String>> getActualFeedbacks(int accountId, int count, String startFrom);

    Single<AnswerVKOfficialList> getOfficial(int accountId, Integer count, Integer startFrom);

    Completable maskAaViewed(int accountId);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/domain/IFeedbackInteractor.kt
package dev.ragnarok.fenrir.domain

import dev.ragnarok.fenrir.model.FeedbackVKOfficialList
import dev.ragnarok.fenrir.model.feedback.Feedback
import dev.ragnarok.fenrir.util.Pair
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface IFeedbackInteractor {
    fun getCachedFeedbacks(accountId: Int): Single<List<Feedback>>
    fun getActualFeedbacks(
        accountId: Int,
        count: Int,
        startFrom: String?
    ): Single<Pair<List<Feedback>, String?>>

    fun getOfficial(accountId: Int, count: Int?, startFrom: Int?): Single<FeedbackVKOfficialList>
    fun markAsViewed(accountId: Int): Single<Boolean>
    fun hide(accountId: Int, query: String?): Completable
}